### PR TITLE
CDAP-19650: add fqn to ftp

### PR DESCRIPTION
**Pull request by mrahanjam**
This PR adds FQN support to the GCS plugins.

Jira: [CDAP-19650](https://cdap.atlassian.net/browse/CDAP-19650)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@
     <data.pipeline.parent>system:cdap-data-pipeline[6.1.2-SNAPSHOT,7.0.0-SNAPSHOT)</data.pipeline.parent>
     <data.stream.parent>system:cdap-data-streams[6.1.2-SNAPSHOT,7.0.0-SNAPSHOT)</data.stream.parent>
     <cdap.version>6.4.0</cdap.version>
-    <hydrator.version>2.6.0</hydrator.version>
+    <hydrator.version>2.10.0-SNAPSHOT</hydrator.version>
```
Comment (yaojiefeng) [inline pom.xml:40]: This should be 2.7.0-SNAPSHOT

**Review by yaojiefeng — APPROVED** [review]
Minor comment, LGTM

File: pom.xml (hunk 1 above)
Comment (mrahanjam) [reply pom.xml:40]: the latest changes needed her are in 2.10.0-SNAPSHOT.

File: src/main/java/io/cdap/plugin/batch/source/FTPBatchSource.java [hunk 2]
```diff
@@ -67,12 +70,24 @@ public class FTPBatchSource extends AbstractFileSource {
...
 
+  @Override
+  public void prepareRun(BatchSourceContext context) throws Exception {
+    // create asset for lineage
+    String referenceName = Strings.isNullOrEmpty(config.getReferenceName())
+      ? ReferenceNames.normalizeFqn(config.getPath())
+      : config.getReferenceName();
+    asset = Asset.builder(referenceName)
+      .setFqn(config.getPath()).build();
+    super.prepareRun(context);
```
Comment (fernst) [inline src/main/java/io/cdap/plugin/batch/source/FTPBatchSource.java:90]: The super call should be first.

**Review by fernst — CHANGES_REQUESTED** [review]

**Review by fernst — APPROVED** [review]

**Pull request merged**